python/lsst/ts/ofc/state_estimator.py

```diff
@@ -190,10 +189,13 @@ def dof_state(
         # With the truncated sensitivity matrix, now include noise covariance
         # when computing the pseudo-inverse. We use the 1e-9 default rcond.
         full_idx = np.concatenate(
-            [self.ofc_data.zn_idx + i * len(self.ofc_data.zn_idx) for i in range(len(sensor_names))]
+            [
+                self.ofc_data.zn_idx + i * (self.ofc_data.znmax - self.ofc_data.znmin + 1)
+                for i in range(len(sensor_names))
+            ]
         )
         used_noise_covariance = noise_covariance_eval[np.ix_(full_idx, full_idx)]
-        noise_cov_inv_sqrt = fractional_matrix_power(used_noise_covariance, -0.5)
+        noise_cov_inv_sqrt = np.linalg.cholesky(used_noise_covariance)
```

Comment:
Isn't this effectively just the square root of the matrix? Don't you then need to invert the Cholesky matrix? Maybe this is why the test is failing?